iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/LocationShifter.java

```diff
@@ -77,6 +77,31 @@ void addLineSize(InputFileContext ctx, int originalLine, int size) {
     getOrCreateLinesShifting(ctx).originalLinesSizes.put(originalLine, size);
   }
 
+  /**
+   * It calculates shifted location in 3 steps:<br/>
+   * <ul>
+   *   <li>see {@link LocationShifter#computeShiftedLocation(InputFileContext, TextRange)}</li>
+   *   <li>see {@link LocationShifter#computeHelmValuePathTextRange(HelmInputFileContext, TextRange)}</li>
+   *   <li>if the location from 1st and 2nd step is the same then first line location is taken and line offsets of original TextRange</li>
+   * </ul>
+   */
+  public TextRange shiftLocation(InputFileContext currentCtx, TextRange textRange) {
+    var shiftedToLine = computeShiftedLocation(currentCtx, textRange);
+    var shiftedTextRange = shiftedToLine;
+    if (currentCtx instanceof HelmInputFileContext helmContext) {
+      shiftedTextRange = computeHelmValuePathTextRange(helmContext, shiftedToLine);
+    }
+    if (shiftedTextRange.equals(shiftedToLine)) {
```

Comment:
I wonder if there can be a case like
```yaml
{{ print "foo: "
-}} bar
```
After evaluation we get
```yaml
foo: bar
#    ^^^
```
but after going back to original offsets we will get
```yaml
{{ print "foo: "
-}} bar
#    ^^^
```
because part of the line is occupied by template and thus will shift.

Reply:
There are a lot of edge cases. Let's verify it later.